src/components/document-content.tsx

```diff
@@ -31,6 +31,18 @@ export class DocumentContentComponent extends BaseComponent<IProps, IState> {
   public state: IState = {};
 
   private domElement: HTMLElement | null;
+  private mutationObserver: MutationObserver;
```

Comment:
Never used something like this before! Would it have been similar to do something like

```
onComponentDidUpdate(newProps) {
  // check if there are more rows
}
```

Reply:
I presume that would have worked, too. The MutationObserver has the advantage that it tells you what's change, i.e. what the new row(s) are, whereas we'd have to figure that out for ourselves in componentDidUpdate(). Since the scroll position isn't managed through React, the work has to be done at the DOM level, so doing it completely at the DOM level made as much sense as doing it partially in React and partially at the DOM level.

Reply:
I guess having all the new nodes is pretty handy

Reply:
Ah I missed your comment before I made mine but that all makes sense!